clouddriver/clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/Main.groovy

```diff
@@ -84,10 +87,19 @@ class Main extends SpringBootServletInitializer {
 
   @Bean
   @Primary
-  ObjectMapper objectMapper(Jackson2ObjectMapperBuilder builder) {
-    return builder.createXmlMapper(false)
-      .mixIn(Artifact.class, ArtifactMixin.class)
-      .build()
+  ObjectMapper objectMapper(Jackson2ObjectMapperBuilder builder, Optional<DeserializerHookRegistry> deserializerHook, Optional<SerializerHookRegistry> serializerHook) {
+    builder = builder.createXmlMapper(false)
+      .mixIn(Artifact.class, ArtifactMixin.class);
+
+    SimpleModule module = new SimpleModule("registryHook")
+    if (deserializerHook.isPresent()) {
+      module.setDeserializerModifier(deserializerHook.get())
+    }
+    if (serializerHook.isPresent()) {
+      module.setSerializerModifier(serializerHook.get());
+    }
+    builder.modules(l -> l.add(module))
```

Comment:
Is it necessary to add the module if neither hook is present?

Reply:
No, but I did feel like this was easier to read. Given this is a bean and only is done once at startup, I did not think I need to wrap it in an if. However, if you feel strongly about it, I can change it

Reply:
I'm not sure of the consequences of having more modules in the ObjectMapper...My reflex is that performance will be the best with only the modules we really need, but I don't have data.  I feel like it's worth an if statement though.